docs/RFCS/txn_gc.md

```diff
@@ -0,0 +1,62 @@
+  transaction is initiated on via `BeginTransaction`).
+* there is exactly one sequence cache entry for each `Range` mutated by the txn
+  with a sequence counter (mutation with non-increasing sequence number triggers
+  a txn restart).
+* in the absence of a transaction record, a `Push` always aborts the transaction.
+
+## GCing Txn Records
+
+The above means that we can always garbage collect aborted transactions with only a best-effort attempt to clean up their intents (but we'll do it only after the client's `EndTransaction` or, if that never happens, the "slow way" via the GC queue; see below).
+For committed transactions, we must guarantee that no open intents exist before deleting the entry (we already synchronously resolve all intents local to the transaction record and GC the record right away if no external intents exist). The straightforward solution is to have `EndTransaction` persist the external intents on the transaction record and let the goroutine which resolves them asynchronously do a little more work: after successfully carrying out the batch worth of `ResolveIntent`, it can delete the corresponding txn record.
+All of this is best effort: we're still going to have a gc queue which walks over old transaction entries, poking old transactions and retrying their intent resolution for the .0001% of transactions which are left hanging.
+
+## GCing Sequence Cache Entries
+
+* it's safe to remove a sequence cache entry when we know the transaction isn't running any more. That's after the client's `EndTransaction` gets executed (regardless of its outcome), but not when a concurrent transaction manages to abort it by means of a `Push`.
+* there are sequence cache entries on any range mutated by the transaction, and we'll be sending `ResolveIntent` there anyways, so we simply make clearing (idempotently) the sequence cache entry a side effect of a `ResolveIntent{,Range}` (when it's carried out as part of `EndTransaction`).
+
+## Intentional Sequence Cache Poisoning
+
+on `ResolveIntent` triggered through an aborting `Push`, we can actually deal with #2231 nicely. The issue there is that a running transaction may not know that it's been aborted already, which leads to anomalies related to the fact that its intents may be gone (so it may not read what it wrote). The key, again, is `ResolveIntent{,Range}`:
+
+* store not only the sequence number, but also the minimal expected epoch.
+* `(epoch, seq) < (epoch', seq') iff epoch < epoch' || (epoch == epoch' && seq < seq')`.
+* upon aborting an intent after a `Push`, we simply poison the sequence cache on that range (setting `sequence=math.MaxInt64`). Assuming that we check the sequence cache on **every** batch (not only for writes), we trigger a transaction restart should the transaction come back to the `Range`. If checking the sequence cache on reads shows up in performance considerations, there are going to be ways to avoid disk I/O in most cases.
```

Comment:
they're GC'ed in the same way as any other. We poison when the transaction is aborted, but the client will (almost always) eventually still send its `EndTransaction` with its intents, triggering (possibly redundant) `ResolveIntent{,Range}` requests which finally delete the entries.